sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BeamSqlUnparseContext.java

```diff
@@ -147,4 +165,35 @@ public int hashCode() {
       return super.hashCode();
     }
   }
+
+  private static class NamedDynamicParam extends SqlDynamicParam {
```

Comment:
nit: AutoValue might save you some boilerplate (equals, hashcode) in this class.

Reply:
Actually I don't think `equal` and `hashcode` is used at all.